src/rgw/driver/sfs/object.cc

```diff
@@ -23,15 +23,15 @@ using namespace std;
 namespace rgw::sal {
 
 SFSObject::SFSReadOp::SFSReadOp(SFSObject *_source)
-    : source(_source) {}
+    : source(_source) {
+  source->refresh_meta();
+  objref = source->get_object_ref();
+}
 
 int SFSObject::SFSReadOp::prepare(
   optional_yield y,
   const DoutPrefixProvider *dpp
 ) {
-
-  source->refresh_meta();
-  objref = source->get_object_ref();
```

Comment:
what's the reasoning behind moving this to the constructor?

Reply:
I cannot remember the exact case, but keeping this into `prepare` is not sufficient in all cases.
There are pieces of SAL code that are calling `get_*()` methods but they don't call `prepare()`.
In these cases the  `SFSReadOp` is not properly initialized and those calls are going to fail.
As solution to this I moved that initializing code into the constructor to cover all the cases.

Reply:
ack. Mind adding a comment on the constructor saying that?